pyms/utils/crypt.py

```diff
@@ -0,0 +1,57 @@
+import base64
+import os
+from typing import Text
+
+from cryptography.fernet import Fernet
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
+
+from pyms.constants import CRYPT_FILE_KEY_ENVIRONMENT, DEFAULT_KEY_FILENAME
+from pyms.exceptions import FileDoesNotExistException
+from pyms.utils.files import LoadFile
+
+
+class Crypt:
+    def __init__(self, *args, **kwargs):
+        self._loader = LoadFile(kwargs.get("path"), CRYPT_FILE_KEY_ENVIRONMENT, DEFAULT_KEY_FILENAME)
+
+    def generate_key(self, password: Text, write_to_file: bool = False):
+        password = password.encode()  # Convert to type bytes
+        salt = os.urandom(16)
+        kdf = PBKDF2HMAC(
+            algorithm=hashes.SHA256(),
```

Comment:
What would you think about using SHA-512/256? It seems that in some edge cases is more secure. https://en.wikipedia.org/wiki/Secure_Hash_Algorithms
It also seems to be supported out of the box. https://cryptography.io/en/latest/hazmat/primitives/cryptographic-hashes/#cryptography.hazmat.primitives.hashes.SHA512_256

Reply:
Changed, Thanks! :-1: